packages/devextreme/js/__internal/scheduler/appointment_popup/appointment_popup.test.ts

```diff
@@ -0,0 +1,952 @@
+  const target = radios.find((radio) => {
+    const label = radio.getAttribute('aria-label')?.trim();
+    const text = radio.textContent?.trim();
+    return label === value || text === value;
+  });
+
+  if (!target) throw new Error(`Radio with value "${value}" not found`);
+
+  radios.forEach((r) => {
+    r.setAttribute('aria-checked', 'false');
+    r.classList.remove('dx-item-selected', 'dx-radiobutton-checked');
+  });
+
+  target.setAttribute('aria-checked', 'true');
+  target.classList.add('dx-item-selected', 'dx-radiobutton-checked');
+
+  target.dispatchEvent(new MouseEvent('click', { bubbles: true }));
+
+  return target;
+}
+
+function getSelectedRadio(container: HTMLElement): HTMLElement | null {
+  return container.querySelector('[role="radio"][aria-checked="true"]');
+}
+
+function getSelectedRadioValue(container: HTMLElement): string | null {
+  const selected = getSelectedRadio(container);
+  return selected?.getAttribute('aria-label') ?? selected?.textContent?.trim() ?? null;
+}
+
+describe('Appointment popup form', () => {
+  beforeEach(() => {
+    fx.off = true;
+    setupSchedulerTestEnvironment();
+  });
+
+  afterEach(() => {
+    fx.off = false;
+    document.body.innerHTML = '';
+    jest.useRealTimers();
+  });
+
+  it('Original appointment\'s fields shouldn\'t fill if used fieldExpr', async () => {
+    const data: Record<string, unknown>[] = [];
+    const textExpValue = 'Subject';
+
+    let newAppointment: Record<string, unknown> | null = null;
+
+    const { scheduler, POM } = await createScheduler({
+      dataSource: data,
+      views: ['week'],
+      currentView: 'week',
+      currentDate: new Date(2021, 4, 27),
+      textExpr: textExpValue,
+      onAppointmentAdded: ({ appointmentData }) => {
+        newAppointment = appointmentData;
+      },
+      height: 600,
+    });
+
+    scheduler.showAppointmentPopup();
+
+    const form = POM.getForm() as HTMLElement;
+    setInputValueByLabel('Subject', 'qwerty', form);
+
+    POM.clickDoneButton();
+
+    expect(newAppointment?.[textExpValue]).toBe('qwerty');
+    // @ts-expect-error set the value in the promise
+    expect(newAppointment?.text).toBeUndefined();
+
+    expect(data[0].Subject).toBe('qwerty');
+    expect(data[0].text).toBeUndefined();
+  });
+
+  it('Recurrence form should work properly if recurrenceRule property mapped recurrenceRuleExpr', async () => {
+    const { scheduler, POM } = await createScheduler({
+      dataSource: [{
+        text: 'Watercolor Landscape',
+        startDate: new Date(2017, 4, 1, 9, 30),
+        endDate: new Date(2017, 4, 1, 11),
+        customRecurrenceRule: 'FREQ=WEEKLY;BYDAY=TU,FR;COUNT=10',
+      }],
+      views: ['month'],
+      currentView: 'month',
+      currentDate: new Date(2017, 4, 25),
+      recurrenceRuleExpr: 'customRecurrenceRule',
+      height: 600,
+    });
+
+    openPopupByDblClick(POM);
+
+    const editSeriesButton = document.querySelector('[aria-label="Edit series"]') as HTMLElement;
```

Comment:
You use it several times. Let's create small object model for this dialog. As in testcafe https://github.com/DevExpress/DevExtreme/blob/a7e5a8bdcf7a6e420185e2813c248abc451a9271/packages/testcafe-models/scheduler/appointment/dialog.ts